modules/nlu/src/backend/predict-pipeline.ts

```diff
@@ -439,6 +451,16 @@ async function extractSlots(input: PredictStep, predictors: Predictors): Promise
   return { ...input, slot_predictions_per_intent: slots_per_intent }
 }
 
+async function extractElectedIntentSlots(input: PredictOutput): Promise<PredictOutput> {
+  const intentWasElectedWithoutAmbiguity = input?.intent?.name && !_.isEmpty(input.predictions) && !input.ambiguous
+  if (!intentWasElectedWithoutAmbiguity) {
+    return input
+  }
+
+  const electedIntent = _.flatMap(input.predictions, p => p.intents).find(i => i.label === input.intent.name) // might find an intent with the correct name of an incorrect context... Chosen context should maybe be in PredictOuput
+  return { ...input, slots: electedIntent.slots } // no more range validation (in  utterance.tagSlot())
```

Comment:
In the previous code, slots of elected intent where tagged in the utterance object using tagSlot() method and where then mapped in MapStepToOutput() function.

In tagSlot() there used to be a range validation ; a check that verify the slots start and end are valid.

Should we do a similar check here? The check will eventually be in a hook.